src/main/java/com/fasterxml/jackson/databind/ObjectMapper.java

```diff
@@ -3846,8 +3846,6 @@ public <T> T readValue(String content, JavaType valueType)
         _assertNotNull("content", content);
         try { // since 2.10 remove "impossible" IOException as per [databind#1675]
             return (T) _readMapAndClose(_jsonFactory.createParser(content), valueType);
-        } catch (JsonProcessingException e) {
```

Comment:
`JsonProcessingException` is an `IOException`, so this block avoids unnecessarily creating a `JsonMappingException` when the `JsonProcessingException` is already descriptive

Reply:
Exactly.